src/ccache/util/file.cpp

```diff
@@ -66,6 +66,10 @@
 #  include <windows.h>
 #endif
 
+#ifdef __linux__
+#  include <sys/sendfile.h>
+#endif
```

Comment:
There are more systems than Linux that have `sys/sendfile.h`. I suggest adding a check in `cmake/GenerateConfigurationFile.cmake` + `cmake/config.h.in` and then use `#ifdef HAVE_SYS_SENDFILE_H` instead.